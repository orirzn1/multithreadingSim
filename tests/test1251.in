L13
S6
O12
N6

T0
I@0x00000000
STORE $4, $1, 0x4
SUB $3, $2, $5
LOAD $0, $5, $3
ADDI $7, $2, 0x2
LOAD $4, $7, 0x1
LOAD $4, $7, 0x4
SUB $2, $7, $1
STORE $3, $4, 0x3
STORE $6, $0, $0
HALT $0

T1
I@0x00000000
ADDI $5, $6, 0x4
SUBI $1, $4, 0x1
ADD $2, $1, $5
LOAD $0, $1, 0x1
HALT $0

T2
I@0x00000000
SUBI $1, $2, 0x3
LOAD $1, $2, 0x2
SUBI $5, $7, 0x2
SUBI $6, $3, 0x1
HALT $0

T3
I@0x00000000
STORE $3, $5, 0x4
ADD $3, $1, $3
ADDI $5, $6, 0x1
ADDI $4, $1, 0x1
STORE $3, $3, $3
ADDI $1, $4, 0x2
LOAD $4, $4, $3
STORE $2, $7, 0x1
HALT $0

T4
I@0x00000000
LOAD $0, $6, $2
ADDI $1, $7, 0x3
STORE $4, $7, $7
LOAD $5, $0, 0x2
STORE $4, $7, 0x4
SUBI $1, $0, 0x1
ADD $4, $0, $1
STORE $6, $2, 0x0
STORE $0, $4, 0x4
HALT $0

T5
I@0x00000000
SUB $0, $5, $0
LOAD $3, $4, 0x1
ADDI $0, $6, 0x1
ADD $2, $7, $7
STORE $3, $2, 0x1
SUB $4, $5, $0
ADD $2, $4, $1
SUB $4, $5, $3
HALT $0

D@0x0
0x6
0x7
0x5
0x8
0x5
0x2
0x8
