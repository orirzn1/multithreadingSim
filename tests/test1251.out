
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x6	R1 = 0x0	R2 = 0x2	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 1:
	R0 = 0x6	R1 = 0xFFFFFFFF	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x6	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x6	R1 = 0x5	R2 = 0x0	R3 = 0x0	R4 = 0xB	R5 = 0x5	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x1	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x6	R4 = 0xFFFFFFFA	R5 = 0x0	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 7.250000

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 1:
	R0 = 0x0	R1 = 0xFFFFFFFF	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0xFFFFFFFF	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x1	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.604167

